activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -14,6 +14,7 @@ def initialize(duration, precision: nil)
       def serialize
         output = 'P'
         parts, sign = normalize
+        return output << "T0S" if parts.empty?
```

Comment:
This combination of return, side effects, and guard clause all at once really bothers me. I'd prefer it if we could structure it as:

```
parts, sign = normalize
return "PT0S".freeze if parts.empty?

output = 'P'
# ...
```